### DISCUSSION TITLE
self-hosting docker run just stops, no error

### DISCUSSION THREAD

**Discussion by bmaginni** [renovatebot/renovate#21864] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.61.0

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

I'm running as self-hosted in docker container as per the self hosting documentation.
It works perfect and produces what I need when I restrict the subset of packages using  "matchPackagePrefixes".
However when I expand to include all the packages I need, it processes for a while and then just stops and exits(255). No error.
Last logs are just "Looking up....."
I'd like some help determining why this is happening

### What you tried so far.

I have DEBUG log level on.
I have tried increasing the memory in docker to 10GB
I have tried increasing memory available to node using -e "NODE_OPTIONS=--max-old-space-size=4096" in docker run command
I have tried reducing memory available to node using -e "NODE_OPTIONS=--max-old-space-size=500" in docker run command and then I do see OOM
I have tried increasing the hostRules timeout to 960000, keepalive:true

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Found 47 new releases for com.*******.service.****-api:****-core in repository https://mavenmaster.trusted.host.com/repository/COMP_Snapshot_Group/ (repository=CM/repo-api, baseBranch=feature/TLPFEAT-130)
DEBUG: Looking up com.********.service.*********-api:additionalServices-core in repository https://mavenmaster.trusted.host.com/repository/COMP_Snapshot_Group/ (repository=CM/repo-api, baseBranch=feature/TLPFEAT-130)

```

</details>

**Comment by rarkins** [comment]
Most likely the container is getting killed by the host, one way or another - e.g. OoM. Try reducing your dependency on the host until you can remove the problem, or get better debugging. For example if you're running it in k8s, stop that and do a regular docker run.

Next, try starting the container manually to /bin/bash and then running `renovate` from within it manually. Then you can hopefully see whether the `renovate` process exits (in which case you'll be returned to the container prompt) or the whole container is exiting, taking Renovate with it.

**Comment by bmaginni** [comment]
Thanks for the reply.
I have tried running it interactively as you say
docker run -m 3GB -it -v "/git/renovate/renovate-config.js:/usr/src/app/config.js" -e LOG_LEVEL=debug -e "NODE_OPTIONS=--max-old-space-size=4096"  bm/renovate /bin/bash
then calling renovate from prompt.
I was returned to the container prompt so container remains up.
Is there any additional logging I can leverage?

**Reply by rarkins** [reply]
Unfortunately it seems like an environmental problem or at the least cannot be debugged/determined from remote, so there's not too much else we can guide on

**Reply by viceice** [reply]
you probably need to increase the allowed memory to the container. or simply disable limit for tests and monitor usage

**Reply by bmaginni** [reply]
Is there any way to get better or more detailed logging to help track down issues?

**Reply by rarkins** [reply]
Are you already running with LOG_LEVEL=debug?

**Reply by bmaginni** [reply]
I am running with LOG_LEVEL=debug.

I have increased memory for container to 10GB and for node to 4GB via (--max-old-space-size=4096)

I have been trying to get node diagnostics using following from command prompt on container
node --report-uncaught-exception --report-on-fatalerror --report-on-signal --report-signal=SIGUSR2 --report-filename=./report.json --report-directory=/home /opt/buildpack/tools/renovate/35.61.0/node_modules/renovate/dist/renovate.js
Not generating a report so far...

But if you guys know of any other good ways of getting more info, would be appreciated

**Reply by viceice** [reply]
run `docker stats <renovate>` in second terminal and view memory usage?

**Comment by bmaginni** [comment]
Interestingly, from the container prompt that was returned, I kept calling renovate cmd again.
It exited to prompt without completing several more times but then it did eventually finish...
